### PR TITLE
Machine settings debug logging and knobs

### PR DESCRIPTION
**Pull request by JoschD**
- No knobs by default
- 'all' and 'default' options
- more debug messages

### CODE REVIEW

File: pylhc/data_extract/lsa.py [hunk 1]
```diff
@@ -134,19 +140,22 @@ def find_beamprocess_history(
         """
         cts = self.findUserContextMappingHistory(t_start.timestamp(), t_end.timestamp(), accelerator=accelerator)
 
-        db = pytimber.LoggingDB(source=source)
+        db = pytimber.LoggingDB(source=source, loglevel=logging.ERROR)
```
Comment (fsoubelet) [inline pylhc/data_extract/lsa.py:143]: nice 👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
 # pylhc Changelog
 
+## Version 0.7.4
+
+Changes in Machine Settings Info
+ - Default behaviour for no knobs given: extract None.
+ - Old behaviour of extracting all restored by giving `knobs = ["all"]`
+ - Option `["default"]` available for default knobs.
```
Comment (fsoubelet) [inline CHANGELOG.md:8]: should be without the brackets, shouldn't it? Or does it refer to what should be passed to the API (not commandline)?

File: pylhc/machine_settings_info.py [hunk 3]
```diff
@@ -198,13 +210,21 @@ def get_info(opt) -> Dict[str, object]:
     except ValueError as e:
         LOG.error(str(e))
     else:
-        trim_histories = LSA.get_trim_history(beamprocess_info.Object, opt.knobs, start_time=acc_start_time, end_time=acc_time, accelerator=opt.accel)
-        trims = _get_last_trim(trim_histories)
+        if opt.knobs is not None:
+            if len(opt.knobs) == 1 and opt.knobs[0].lower() == 'all':
+                opt.knobs = []  # will extract all knobs in get_trim_history
+            if len(opt.knobs) == 1 and opt.knobs[0].lower() == 'default':
```
Comment (fsoubelet) [inline pylhc/machine_settings_info.py:216]: `elif` maybe, not a big deal

File: pylhc/machine_settings_info.py [hunk 4]
```diff
@@ -198,13 +210,21 @@ def get_info(opt) -> Dict[str, object]:
...
+                opt.knobs = [name2lsa(knob) for category in KNOB_CATEGORIES.values()
+                             for knob in category]
+
+            trim_histories = LSA.get_trim_history(beamprocess_info.Object, opt.knobs, start_time=acc_start_time, end_time=acc_time, accelerator=opt.accel)
+            trims = _get_last_trim(trim_histories)
+
         if opt.knob_definitions:
-            if opt.knobs:
-                knob_definitions = _get_knob_definitions(opt.knobs, optics_info.Name)
+            if trim_histories:  # this now works with 'all'. Might create a lot of files
```
Comment (fsoubelet) [inline pylhc/machine_settings_info.py:224]: also with 'default', no?

File: pylhc/data_extract/lsa.py (hunk 1 above)
Comment (fsoubelet) [reply pylhc/data_extract/lsa.py:143]: On second thought, maybe we want the warnings too?

**Review by fsoubelet — COMMENTED** [review]
Few clarifications needed for my tired brain 😬

File: pylhc/machine_settings_info.py (hunk 3 above)
Comment (JoschD) [reply pylhc/machine_settings_info.py:216]: yeah. had it first, but code looks nicer this way (I think we can accept the inefficiency)

File: pylhc/machine_settings_info.py (hunk 4 above)
Comment (JoschD) [reply pylhc/machine_settings_info.py:224]: true

**Review by fsoubelet — APPROVED** [review]

**Pull request merged**